oddt/surface.py

```diff
@@ -0,0 +1,166 @@
+    plot_atoms : bool (default = True)
+        If True, also shows a scatterplot of atomic coordinates of the molecule.
+    alpha : float (default = 1)
+        A transparency parameter, the lower it is set the more transparent the plot will be.
+    color : (red, green, blue) (default = (0, 0, 1))
+        Color of the plotted surface in the RGB format.
+    figure_size : (X, Y) (default = (10, 10))
+        Size of the displayed plot.
+    remove_hoh : bool (default = False)
+        If True, remove waters from the molecule before generating the surface
+    probe_radius : float (default = 1.4)
+        Radius of a ball used to patch up holes inside the molecule resulting from
+        some molecular distances being larger (usually in protein). Basically reduces
+        the surface to one accesible by other molecules of radius smaller than probe_radius.
+
+    Returns
+    -------
+    None
+
+    """
+    # Calculate plot axis boundries 
+    no_hoh = molecule.atom_dict['resname'] != 'HOH'
+    coords = molecule.atom_dict['coords'][no_hoh]
+    radius = molecule.atom_dict['radius'][no_hoh].max()
+    xmiddle, ymiddle, zmiddle = np.median(coords, axis=0)
+    xlim, ylim, zlim = coords.max(axis=0) - coords.min(axis=0) + 5 * radius
+    # Marching cubes
+    verts, faces = generate_surface_marching_cubes(molecule, scaling=scaling,
+                                                   remove_hoh=remove_hoh, probe_radius=probe_radius)
+    # Create the plot
+    fig = plt.figure(figsize=figure_size)
+    ax = fig.add_subplot(111, projection='3d')
+    mesh = Poly3DCollection(verts[faces], linewidth=1, facecolors=(*color, alpha), alpha=alpha)
+    if plot_atoms:
+        ax.scatter(*coords.T)
+    mesh.set_edgecolor('k')
+    ax.add_collection3d(mesh)
+    ax.set_xlim(xmiddle - xlim/2, xmiddle + xlim/2)
+    ax.set_ylim(ymiddle -ylim/2, ymiddle + ylim/2)
+    ax.set_zlim(zmiddle - zlim/2, zmiddle + zlim/2)
+    plt.tight_layout()
+    plt.show()
+
+def find_surface_residues(molecule, max_dist=None, scaling=1):
+    """Finds residues close to the molecular surface using generate_surface_marching_cubes.
+
+    Parameters
+    ----------
+    molecule : oddt.toolkit.Molecule
+        Molecule to find surface residues in.
+    max_dist : float (default = None)
+        Maximum distance from the surface where residues would still be considered close.
+        If None, compares distances to radii of respective atoms.
+    scaling : float (default = 1)
+        Expands the grid in which computation is done by generate_surface_marching_cubes
+        by a factor of scaling. Results in a more accurate representation of the surface,
+        and therefore more accurate computation of distances but increases computation time.
+
+    Returns
+    -------
+    atom_dict : numpy array
+        An atom_dict containing only the surface residues from the original molecule.
+    """
+    # Copy the atom_dict, remove waters
+    atom_dict = molecule.atom_dict.copy()
```

Comment:
Why do we need to copy `atom_dict`? If there is a reason - write it in the comment.